snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -921,39 +926,23 @@ def _clean_shared_area(self, shared_directory, part_state, project_state):
         # part.
         _clean_migrated_files(primed_files, primed_directories, shared_directory)
 
-    def _handle_dependencies(
-        self, *, all_dependencies: Set[str], content_dirs: Set[str]
-    ):
-        # Split the necessary dependencies into their corresponding location.
-        # We'll only track the part and staged dependencies, since they should have
-        # already been primed by other means, and migrating them again could
-        # potentially override the `stage` or `snap` filtering.
-        dirs = [self.plugin.installdir, self.stagedir, self.primedir, *content_dirs]
+    def _calculate_dependency_paths(
+        self, split_depencies: Dict[str, Set[str]]
+    ) -> Set[str]:
+        part_dependencies: Set[str] = split_depencies.get(self.primedir, set())
```

Comment:
mypy should be able to infer this type, no?

Reply:
Ah yes good point.